File: app/src/main/java/com/xadaptersimple/TestActivity.java
package com.xadaptersimple;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.xadapter.adapter.XRecyclerViewAdapter;
import com.xadapter.holder.XViewHolder;
import com.xadapter.listener.LoadListener;
import com.xadapter.listener.OnXBindListener;
import com.xadapter.widget.SimpleLoadMore;
import com.xadapter.widget.SimpleRefresh;
import com.xadaptersimple.data.DataUtils;
import com.xadaptersimple.data.MainBean;

import java.util.ArrayList;
import java.util.List;

/**
 * by y on 2017/6/20.
 */

public class TestActivity extends AppCompatActivity implements OnXBindListener<MainBean>, LoadListener {

    private XRecyclerViewAdapter<MainBean> xRecyclerViewAdapter;
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.recyclerview_layout);
        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        List<MainBean> mainBeen = new ArrayList<>();
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        xRecyclerViewAdapter = new XRecyclerViewAdapter<>();
        recyclerView.setAdapter(
                xRecyclerViewAdapter
                        .initXData(mainBeen)
                        .addRecyclerView(recyclerView)
                        .setLayoutId(R.layout.item)
                        .setPullRefreshEnabled(true)
                        .setLoadingMoreEnabled(true)
                        .onXBind(this)
                        .setLoadListener(this)
                        .refresh()
        );

    }

    @Override
    public void onXBind(XViewHolder holder, int position, MainBean mainBean) {
        holder.setTextView(R.id.tv_name, mainBean.getName());
        holder.setTextView(R.id.tv_age, mainBean.getAge() + "");
    }

    @Override
    public void onRefresh() {
        xRecyclerViewAdapter.removeAll();
        recyclerView.postDelayed(new Runnable() {
            @Override
            public void run() {
                xRecyclerViewAdapter.addAllData(DataUtils.getTestData(new ArrayList<MainBean>()));
                xRecyclerViewAdapter.refreshState(SimpleRefresh.SUCCESS);
                if (xRecyclerViewAdapter.getData().size() < 7) {
                    xRecyclerViewAdapter.loadMoreState(SimpleLoadMore.NOMORE);
                }
            }
        }, 1500);
    }

    @Override
    public void onLoadMore() {
        recyclerView.postDelayed(new Runnable() {
            @Override
            public void run() {
                if (xRecyclerViewAdapter.getData().size() < 7) {
                    xRecyclerViewAdapter.loadMoreState(SimpleLoadMore.NOMORE);
                } else {
                    xRecyclerViewAdapter.loadMoreState(SimpleLoadMore.ERROR);
                }
            }
        }, 1500);
    }
}

File: app/src/main/java/com/xadaptersimple/TestActivity.kt
package com.xadaptersimple

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.xadapter.adapter.XRecyclerViewAdapter
import com.xadapter.addAll
import com.xadapter.holder.setText
import com.xadapter.refresh
import com.xadapter.removeAll
import com.xadapter.widget.XLoadMoreView
import com.xadapter.widget.XRefreshView
import com.xadaptersimple.data.DataUtils
import com.xadaptersimple.data.MainBean
import kotlinx.android.synthetic.main.recyclerview_layout.*
import java.util.*

/**
 * by y on 2017/6/20.
 */

class TestActivity : AppCompatActivity() {

    private lateinit var xRecyclerViewAdapter: XRecyclerViewAdapter<MainBean>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.recyclerview_layout)
        val mainBeen = ArrayList<MainBean>()
        recyclerView.layoutManager = LinearLayoutManager(this)
        xRecyclerViewAdapter = XRecyclerViewAdapter()
        recyclerView.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
        recyclerView.adapter = xRecyclerViewAdapter
                .apply {
                    dataContainer = mainBeen
                    recyclerView = this@TestActivity.recyclerView
                    itemLayoutId = R.layout.item
                    pullRefreshEnabled = true
                    loadingMoreEnabled = true
                    onXBindListener = { holder, position, entity ->
                        holder.setText(R.id.tv_name, entity.name)
                        holder.setText(R.id.tv_age, entity.age.toString() + "")
                    }
                    xRefreshListener = {
                        xRecyclerViewAdapter.removeAll()
                        this@TestActivity.recyclerView.postDelayed({
                            xRecyclerViewAdapter.addAll(DataUtils.getTestData(ArrayList()))
                            xRecyclerViewAdapter.refreshState = XRefreshView.SUCCESS
                            if (xRecyclerViewAdapter.dataContainer.size < 7) {
                                xRecyclerViewAdapter.loadMoreState = XLoadMoreView.NOMORE
                            }
                        }, 1500)
                    }
                    xLoadMoreListener = {
                        this@TestActivity.recyclerView.postDelayed({
                            if (xRecyclerViewAdapter.dataContainer.size < 7) {
                                xRecyclerViewAdapter.loadMoreState = XLoadMoreView.NOMORE
                            } else {
                                xRecyclerViewAdapter.loadMoreState = XLoadMoreView.ERROR
                            }
                        }, 1500)
                    }
                }
                .refresh()

    }
}
